api/v1alpha1/compression_types.go

```diff
@@ -0,0 +1,28 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type GzipCompressor struct {
+}
+
+type Compression struct {
+	// A compressor library to use for compression
+	//
+	// +required
+	CompressorLibrary *CompressorLibrary `json:"compressorLibrary,omitempty"`
```

Comment:
compressionLibrary adds an extra level, 
prefer
```
compression:
  type: "Gzip"
    gzip:
```

and in the init API maybe we can skip `gzip` struct and rely on defaults if possible

Reply:
got it, thanks!